constants/menuLinks.ts

```diff
@@ -2,10 +2,28 @@ type SegmentData = {
   action: string
 }
 
+/**
+ * To use this type:
```

Comment:
What syntax are you using here?

I would adhere to the JSDoc syntax which I think TSDoc is based. Telling exactly what is the syntax for TSDoc is hard. It is spread across several sources. I've found a good starting point at [this StackOverflow answer](https://stackoverflow.com/questions/23072286/where-is-the-syntax-for-typescript-comments-documented) but I've also found more [examples](https://github.com/microsoft/tsdoc/tree/master/playground/src/samples) in the TSDoc repository.

Looking at how you documented the fields separately, there is an issue in the TSDoc repository about [documenting types](https://github.com/TypeStrong/typedoc/issues/278) that suggests converting the type into an `interface` and document each property. Could you use that style, please? You can describe the type in a general way, i.e. `Represent a navigation link for the menus of the site` (I don't know why, but imperative mood is the usual form here) and then you can go deeper field by field.